core/src/ast/data_type.rs

```diff
@@ -31,4 +31,5 @@ pub enum DataType {
     List,
     Decimal,
     Point,
+    Null,
```

Comment:
Do we really need `DataType::Null`? `NULL` can be the value but it is hard to see the reason to have `NULL` as data type.

Reply:
So the data type of a Null object would be? We can make it so the datatype() method returns an option, with None for that case.

Reply:
ok, I thought that the this pr is for comparing between incompatible types to return error rather than false boolean value. could we remove other elements which are not directly related to the issue? then, it would be good to focus on the core

Reply:
The reason for the introduction of the [`datatype()` method](https://github.com/gluesql/gluesql/blob/3410f789b50e825b01e2e154dcf97608af47d9d6/core/src/data/value/literal.rs#L129-L158) is that it was needed to instantiate [the error when the comparison fails](https://github.com/gluesql/gluesql/blob/3410f789b50e825b01e2e154dcf97608af47d9d6/core/src/data/value/literal.rs#L120-L124)

Reply:
Removed the `Null` data type and introduced new Error messages that do not require `datatype()`